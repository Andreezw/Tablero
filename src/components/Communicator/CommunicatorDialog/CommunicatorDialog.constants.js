export const TAB_INDEXES = {
  COMMUNICATOR_BOARDS: 0,
  ALL_BOARDS: 1,
  MY_BOARDS: 2
};
