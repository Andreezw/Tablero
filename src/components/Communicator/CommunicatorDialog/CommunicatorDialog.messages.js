import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'cboard.components.CommunicatorDialog.title',
    defaultMessage: 'My Communicator'
  },
  menu: {
    id: 'cboard.components.CommunicatorDialog.menu',
    defaultMessage: 'Menu'
  },
  menuRootBoardOption: {
    id: 'cboard.components.CommunicatorDialog.menuRootBoardOption',
    defaultMessage: 'Set as Root Board'
  },
  menuPublishOption: {
    id: 'cboard.components.CommunicatorDialog.menuPublishOption',
    defaultMessage: 'Publish Board'
  },
  menuUnpublishOption: {
    id: 'cboard.components.CommunicatorDialog.menuUnpublishOption',
    defaultMessage: 'Unpublish Board'
  },
  emptyBoardsList: {
    id: 'cboard.components.CommunicatorDialog.emptyBoardsList',
    defaultMessage: 'No boards :('
  },
  loadNextPage: {
    id: 'cboard.components.CommunicatorDialog.loadNextPage',
    defaultMessage: 'Load More'
  },
  search: {
    id: 'cboard.components.CommunicatorDialog.search',
    defaultMessage: 'Search'
  },
  author: {
    id: 'cboard.components.CommunicatorDialog.author',
    defaultMessage: 'By {author}'
  },
  addBoard: {
    id: 'cboard.components.CommunicatorDialog.addBoard',
    defaultMessage: 'Add Board'
  },
  removeBoard: {
    id: 'cboard.components.CommunicatorDialog.removeBoard',
    defaultMessage: 'Remove Board'
  },
  communicatorBoards: {
    id: 'cboard.components.CommunicatorDialog.communicatorBoards',
    defaultMessage: 'Included Boards'
  },
  allBoards: {
    id: 'cboard.components.CommunicatorDialog.allBoards',
    defaultMessage: 'All Boards'
  },
  myBoards: {
    id: 'cboard.components.CommunicatorDialog.myBoards',
    defaultMessage: 'My Boards'
  },
  boardsQty: {
    id: 'cboard.components.CommunicatorDialog.boardsQty',
    defaultMessage: '{qty} boards'
  },
  helpAndSupport: {
    id: 'cboard.components.CommunicatorDialog.helpAndSupport',
    defaultMessage: 'Help And Support'
  },
  termsOfService: {
    id: 'cboard.components.CommunicatorDialog.termsOfService',
    defaultMessage: 'Terms Of Service'
  }
});
