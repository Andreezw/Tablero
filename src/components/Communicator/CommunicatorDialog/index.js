import CommunicatorDialog from './CommunicatorDialog.container';

export default CommunicatorDialog;
