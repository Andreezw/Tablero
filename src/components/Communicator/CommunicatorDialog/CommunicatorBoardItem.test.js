import React from 'react';
import { shallowMatchSnapshot } from '../../../common/test_utils';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CommunicatorBoardItem from './CommunicatorBoardItem.component';
import { TAB_INDEXES } from './CommunicatorDialog.constants';

jest.mock('./CommunicatorDialog.messages', () => {
  return {
    title: {
      id: 'cboard.components.CommunicatorDialog.title',
      defaultMessage: 'My Communicator'
    },
    menu: {
      id: 'cboard.components.CommunicatorDialog.menu',
      defaultMessage: 'Menu'
    },
    search: {
      id: 'cboard.components.CommunicatorDialog.search',
      defaultMessage: 'Search'
    },
    author: {
      id: 'cboard.components.CommunicatorDialog.author',
      defaultMessage: 'By {author}'
    },
    addBoard: {
      id: 'cboard.components.CommunicatorDialog.addBoard',
      defaultMessage: 'Add Board'
    },
    removeBoard: {
      id: 'cboard.components.CommunicatorDialog.removeBoard',
      defaultMessage: 'Remove Board'
    },
    communicatorBoards: {
      id: 'cboard.components.CommunicatorDialog.communicatorBoards',
      defaultMessage: 'Included Boards'
    },
    allBoards: {
      id: 'cboard.components.CommunicatorDialog.allBoards',
      defaultMessage: 'All Boards'
    },
    myBoards: {
      id: 'cboard.components.CommunicatorDialog.myBoards',
      defaultMessage: 'My Boards'
    },
    boardsQty: {
      id: 'cboard.components.CommunicatorDialog.boardsQty',
      defaultMessage: '{qty} boards'
    }
  };
});

const intlMock = {
  formatMessage: ({ id }) => id
};

const COMPONENT_PROPS = {
  intl: intlMock,
  board: {
    id: 'someid',
    author: 'test author',
    nameKey: 'some.namekey.for.board',
    isPublic: false,
    tiles: []
  }
};

describe('CommunicatorBoardItem tests', () => {
  test('default renderer', () => {
    shallowMatchSnapshot(<CommunicatorBoardItem {...COMPONENT_PROPS} />);
  });

  test('menu behavior', () => {
    const wrapper = shallow(<CommunicatorBoardItem {...COMPONENT_PROPS} />);
    let tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();

    let menu = wrapper.find('.board-item-menu').get(0);
    expect(menu.props.open).toBe(false);

    const menuButton = wrapper.find('.board-item-menu-button').at(0);
    menuButton.simulate('click', { currentTarget: 'someElement' });

    menu = wrapper.find('.board-item-menu').get(0);
    expect(menu.props.open).toBe(true);

    tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
