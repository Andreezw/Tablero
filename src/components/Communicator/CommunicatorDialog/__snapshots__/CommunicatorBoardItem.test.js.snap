// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommunicatorBoardItem tests default renderer 1`] = `
<div
  className="CommunicatorDialog__boards__item"
>
  <div
    className="CommunicatorDialog__boards__item__image"
  >
    <div
      className="CommunicatorDialog__boards__item__image__empty"
    >
      <pure(BrokenImage) />
    </div>
    <div
      className="CommunicatorDialog__boards__item__image__button"
    >
      <IconButton
        label="cboard.components.CommunicatorDialog.addBoard"
        onClick={[Function]}
      >
        <pure(AddCircle) />
      </IconButton>
    </div>
  </div>
  <div
    className="CommunicatorDialog__boards__item__data"
  >
    <div
      className="CommunicatorDialog__boards__item__data__button"
    >
      <IconButton
        className="board-item-menu-button board-item-menu-button-someid"
        label="cboard.components.CommunicatorDialog.menu"
        onClick={[Function]}
      >
        <pure(MoreVert) />
      </IconButton>
      <WithStyles(Menu)
        anchorEl={null}
        className="board-item-menu board-item-menu-someid"
        onClose={[Function]}
        open={false}
      >
        <WithStyles(MenuItem)
          onClick={[Function]}
        >
          ...
        </WithStyles(MenuItem)>
      </WithStyles(Menu)>
    </div>
    <div
      className="CommunicatorDialog__boards__item__data__title"
    >
      some.namekey.for.board
    </div>
    <div
      className="CommunicatorDialog__boards__item__data__author"
    >
      cboard.components.CommunicatorDialog.author
    </div>
    <div
      className="CommunicatorDialog__boards__item__data__extra"
    >
      
    </div>
  </div>
</div>
`;

exports[`CommunicatorBoardItem tests menu behavior 1`] = `
<div
  className="CommunicatorDialog__boards__item"
>
  <div
    className="CommunicatorDialog__boards__item__image"
  >
    <div
      className="CommunicatorDialog__boards__item__image__empty"
    >
      <pure(BrokenImage) />
    </div>
    <div
      className="CommunicatorDialog__boards__item__image__button"
    >
      <IconButton
        label="cboard.components.CommunicatorDialog.addBoard"
        onClick={[Function]}
      >
        <pure(AddCircle) />
      </IconButton>
    </div>
  </div>
  <div
    className="CommunicatorDialog__boards__item__data"
  >
    <div
      className="CommunicatorDialog__boards__item__data__button"
    >
      <IconButton
        className="board-item-menu-button board-item-menu-button-someid"
        label="cboard.components.CommunicatorDialog.menu"
        onClick={[Function]}
      >
        <pure(MoreVert) />
      </IconButton>
      <WithStyles(Menu)
        anchorEl={null}
        className="board-item-menu board-item-menu-someid"
        onClose={[Function]}
        open={false}
      >
        <WithStyles(MenuItem)
          onClick={[Function]}
        >
          ...
        </WithStyles(MenuItem)>
      </WithStyles(Menu)>
    </div>
    <div
      className="CommunicatorDialog__boards__item__data__title"
    >
      some.namekey.for.board
    </div>
    <div
      className="CommunicatorDialog__boards__item__data__author"
    >
      cboard.components.CommunicatorDialog.author
    </div>
    <div
      className="CommunicatorDialog__boards__item__data__extra"
    />
  </div>
</div>
`;

exports[`CommunicatorBoardItem tests menu behavior 2`] = `
<div
  className="CommunicatorDialog__boards__item"
>
  <div
    className="CommunicatorDialog__boards__item__image"
  >
    <div
      className="CommunicatorDialog__boards__item__image__empty"
    >
      <pure(BrokenImage) />
    </div>
    <div
      className="CommunicatorDialog__boards__item__image__button"
    >
      <IconButton
        label="cboard.components.CommunicatorDialog.addBoard"
        onClick={[Function]}
      >
        <pure(AddCircle) />
      </IconButton>
    </div>
  </div>
  <div
    className="CommunicatorDialog__boards__item__data"
  >
    <div
      className="CommunicatorDialog__boards__item__data__button"
    >
      <IconButton
        className="board-item-menu-button board-item-menu-button-someid"
        label="cboard.components.CommunicatorDialog.menu"
        onClick={[Function]}
      >
        <pure(MoreVert) />
      </IconButton>
      <WithStyles(Menu)
        anchorEl="someElement"
        className="board-item-menu board-item-menu-someid"
        onClose={[Function]}
        open={true}
      >
        <WithStyles(MenuItem)
          onClick={[Function]}
        >
          ...
        </WithStyles(MenuItem)>
      </WithStyles(Menu)>
    </div>
    <div
      className="CommunicatorDialog__boards__item__data__title"
    >
      some.namekey.for.board
    </div>
    <div
      className="CommunicatorDialog__boards__item__data__author"
    >
      cboard.components.CommunicatorDialog.author
    </div>
    <div
      className="CommunicatorDialog__boards__item__data__extra"
    />
  </div>
</div>
`;
